keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1328 @@
++	TrustAnchorSetName string `json:"trustAnchorSetName"`
++}
+```
+
+#### Trust Anchor Configuration for Aggregated API Server Backends
+
+The `apiregistration.k8s.io` v1 APIServiceSpec object will be extended with a
+new optional field `TrustAnchors` that references a single trust anchor set
+within a single ClusterTrustBundle object.  `TrustAnchors` will be mutually
+exclusive with the existing `CABundle` and `InsecureSkipTLSVerify` fields.
+
+When `TrustAnchors` is specified, kube-apiserver will load the certificates from
+the named trust anchor set of the named ClusterTrustBundle object, an use those
+for validating the certificate presented by the aggregated API server.
+kube-apiserver will follow updates to the ClusterTrustBundle object, and use the
+updated certificates within a few seconds.
+
+```diff
+type APIServiceSpec struct {
+    // ...
+
+    // InsecureSkipTLSVerify disables TLS certificate verification when communicating with this server.
+    // This is strongly discouraged.  You should use the CABundle instead.
+    InsecureSkipTLSVerify bool `json:"insecureSkipTLSVerify,omitempty" protobuf:"varint,4,opt,name=insecureSkipTLSVerify"`
+    // CABundle is a PEM encoded CA bundle which will be used to validate an API server's serving certificate.
+    // If unspecified, system trust roots on the apiserver are used.
+    // +listType=atomic
+    // +optional
+    CABundle []byte `json:"caBundle,omitempty" protobuf:"bytes,5,opt,name=caBundle"`
++	// A reference to a dynamic trust anchor set that will be used to validate
++	// the webhook's server certificate.
++	//
++	// Mutually exclusive with CABundle and InsecureSkipTLSVerify.
++	//
++	// This field is alpha.  Using it requires setting the ClusterTrustBundle
++	// feature gate.
++	//
++	// +optional
++	TrustAnchors *TrustAnchorSetReference `json:"trustAnchors,omitempty"`
+
+    // ...
+}
++
++// TrustAnchorSetReference names a particular trust anchor set within a
++// ClusterTrustBundle object.
++type TrustAnchorSetReference struct {
++	// The ClusterTrustBundle object to refer to.
++	ClusterTrustBundleName string `json:"clusterTrustBundleName"`
++
++	// The individual named trust anchor set within the ClusterTrustBundle.
++	TrustAnchorSetName string `json:"trustAnchorSetName"`
++}
+```
+
+### trustAnchors Projected Volume Source
+
+The trustAnchors projected volume source is responsible for retrieving a
+designated trust anchor set from a signer and publishing it into the container
+filesystem.
+
+All of the new fields described in this section are gated by the
+ClusterTrustBundle kubelet feature gate.  Unless the feature gate is enabled,
+usage of these alpha objects and fields will be rejected by kubelet.
+
+#### Configuration Object Definition
+
+```go
+// TrustAnchorsProjection is the top-level configuration object for
+// instances of the trustAnchors kubelet projected volume source.
+type TrustAnchorsProjection struct {
```

Comment:
Can the name of the projection better align with the name of top level kind (ClusterTrustBundle)?

Reply:
Others have expressed interest in eventually supporting a namespace-scoped TrustBundle; I kept the trustAnchors name to reflect that fact that we are loading trust anchors, whether or not those ultimately come from a ClusterTrustBundle or a namespace-scoped TrustBundle.

Reply:
After having read and digested some of your other comments, I think a better summarization of my thoughts here is that the name of the project should align with the *format* we are projecting into the pod, not necessarily the source of the data (although I think maybe it could capture both, like `ClusterTrustBundlePEMSetProjection`).

But I think it's a little cleaner just to call it something like `PEMTrustAnchorSetProjection`.  The thing we are putting into the pod is a single trust anchor set in PEM bundle format.

Reply:
I've gone ahead and done this.